Lib/test/test_asyncore.py

```diff
@@ -800,6 +801,36 @@ def test_quick_connect(self):
             if t.is_alive():
                 self.fail("join() timed out")
 
+    def test_map_altered_in_loop(self):
```

Comment:
To help other understand this issue, I would move this patch before the actual fix, and mark it as @unittest.expectedFailure so it can be merged even if we don't have a good fix.